### PR TITLE
Allow global selector by ID

### PR DESCRIPTION
**Pull request by adamwight**
Selection by ID is safe, we should allow it.  ~~I believe that's what
was originally intended in this rule?~~

Should the same fix be applied to the `context` selector below?

### CODE REVIEW

**Comment by adamwight** [comment]
> I believe that's what was originally intended in this rule?

This was wrong, I see from the test that "#" was intentional, it's a jQuery < 3.3 thing.

Nevertheless, I'd like to get feedback on allowing "#id" selectors.

**Comment by jdforrester** [comment]
Wikimedia's coding standards have for ages strongly discouraged use of IDs. I don't think putting in effort to make it easier to use them is a good idea.

**Comment by edg2s** [comment]
Per James, I wouldn't want to turn this on in eslint-config-wikimedia, but one could make a case for adding an option to allow ID selectors for other users of this plugin.

**Comment by edg2s** [comment]
One could also enable it on a per-project basis. While we should continue to discourage IDs, giving people an option that is less drastic than turning the rule off completely might be beneficial.

**Comment by jdforrester** [comment]
Fine by me, just don't want to get Adam's hopes up.

File: tests/no-global-selector.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ ruleTester.run( 'no-global-selector', rule, {
 		'$(".selector", context)',
 		'$(".selector", "#")',
 		'$(".selector", [])',
+		'$("#id")',
```
Comment (edg2s) [inline tests/no-global-selector.js]: Per discussion on PR, this should be configurable, e.g. { "allowIds": true }, but default to false

**Pull request merged**